src/System.Windows.Extensions/src/System/Media/SoundPlayer.cs

```diff
@@ -0,0 +1,738 @@
+        // called either when the user is setting the path/stream and we are loading
+        // or when loading took more time than the time out
+        private void CleanupStreamData()
+        {
+            _currentPos = 0;
+            _streamData = null;
+            IsLoadCompleted = false;
+            _lastLoadException = null;
+            _doesLoadAppearSynchronous = false;
+            _copyThread = null;
+            _semaphore.Set();
+        }
+
+        public void Load()
+        {
+            // if we have a file there is nothing to load - we just pass the file to the PlaySound function
+            // if we have a stream, then we start loading the stream sync
+            if (_uri != null && _uri.IsFile)
+            {
+                Debug.Assert(_stream == null, "we can't have a stream and a path at the same time");
+                FileInfo fi = new FileInfo(_uri.LocalPath);
+                if (!fi.Exists)
+                {
+                    throw new FileNotFoundException(SR.SoundAPIFileDoesNotExist, _soundLocation);
+                }
+                IsLoadCompleted = true;
+                OnLoadCompleted(new AsyncCompletedEventArgs(null, false, null));
+                return;
+            }
+
+            LoadSync();
+        }
+
+        private void LoadAndPlay(int flags)
+        {
+            // When the user does not specify a sound location nor a stream, play Beep
+            if (string.IsNullOrEmpty(_soundLocation) && _stream == null)
+            {
+                SystemSounds.Beep.Play();
+                return;
+            }
+
+            if (_uri != null && _uri.IsFile)
+            {
+                // Someone can call SoundPlayer::set_Location between the time
+                // LoadAndPlay validates the sound file and the time it calls PlaySound.
+                // The SoundPlayer will end up playing an un-validated sound file.
+                // The solution is to store the uri.LocalPath on a local variable
+                string localPath = _uri.LocalPath;
+
+                // Play the path - don't use uri.AbsolutePath because that gives problems
+                // when there are whitespaces in file names
+                IsLoadCompleted = true;
+
+                ValidateSoundFile(localPath);
+                Interop.WinMM.PlaySound(localPath, IntPtr.Zero, Interop.WinMM.SND_NODEFAULT | flags);
+            }
+            else
+            {
+                LoadSync();
+                ValidateSoundData(_streamData);
+                Interop.WinMM.PlaySound(_streamData, IntPtr.Zero, Interop.WinMM.SND_MEMORY | Interop.WinMM.SND_NODEFAULT | flags);
+            }
+        }
+
+        private void LoadSync()
+        {
+            Debug.Assert((_uri == null || !_uri.IsFile), "we only load streams");
+
+            // first make sure that any possible download ended
+            if (!_semaphore.WaitOne(LoadTimeout, false))
+            {
+                _copyThread?.Abort();
```

Comment:
This looks problematic.  It seems the code is relying on thread aborts in order to handle cancellation or timeouts or some such thing, but thread aborts don't exist in .NET Core, and this call will throw a PlatformNotSupportedException.  This whole mechanism will need to be re-evaluated, as at the moment it'll be broken on core.